docs/authentication-rdd.md

```diff
@@ -0,0 +1,406 @@
+   - Show browser's WebAuthn authentication dialog
+   - Verify passkey against stored credentials
+   
+2. **Successful Authentication**
+   - Redirect to course content landing page
+   - Set authentication session
+   - Update last sign-in timestamp
+
+**Fallback Option**:
+- Link: "Don't have your passkey? Sign in with email"
+
+#### 2.2 Fallback Authentication (Magic Link)
+**Trigger**: User clicks "Sign in with email" fallback option
+
+**Flow**:
+1. **Email Collection**
+   - Prompt for email address
+   - Validate email exists in system
+   
+2. **Magic Link Dispatch**
+   - Send magic link to email
+   - Link expires after 15 minutes
+   - Link is single-use only
+   
+3. **Magic Link Authentication**
+   - User clicks magic link
+   - Validate link authenticity and expiration
+   - Authenticate user session
+   
+4. **Post-Authentication Flow**
+   - If user has existing passkey: Redirect to course content
+   - If user has no passkey: Prompt for passkey setup (optional)
+   - Redirect to course content landing page
+
+**Error Handling**:
+- Email not found → Redirect to sign-up flow
+- Magic link expired → Offer to resend new link
+- Invalid link → Show error with option to request new link
+
+### 3. Device Management Flow
+
+#### 3.1 New Device Access
+**Scenario**: Existing user wants to access account from new device
+
+**Flow**:
+1. User attempts passkey authentication → No passkey available
+2. User clicks "Sign in with email"
+3. Complete magic link authentication
+4. After successful auth, prompt: "Set up passkey for this device?"
+5. Optional passkey registration for new device
+
+#### 3.2 Email Change Flow
+**Scenario**: User wants to change their email address
+
+**Requirements**:
+- Must authenticate with current passkey OR current email magic link
+- Send verification to new email address
+- Maintain existing passkeys after email change
+- Update email in user profile
+
+## Technical Requirements
+
+### 3.1 Security Requirements
+
+#### Passkey Implementation
+- **Standard**: WebAuthn Level 2 compliance
+- **Authenticator Requirements**: 
+  - Platform authenticators preferred (Touch ID, Face ID, Windows Hello)
+  - Cross-platform authenticators supported (security keys)
+- **User Verification**: Required for all passkey operations
+- **Attestation**: None required (for broader compatibility)
+
+#### Magic Link Security
+- **Expiration**: 15 minutes maximum
+- **Single Use**: Links invalidated after first use
+- **Entropy**: Minimum 256 bits of randomness
+- **Rate Limiting**: Maximum 3 links per email per hour
+- **HTTPS**: All magic links must use HTTPS
+
+#### Session Management
+- **Session Duration**: 30 days for passkey auth, 7 days for magic link auth
+- **Session Storage**: Secure HTTP-only cookies
+- **CSRF Protection**: Required for all authenticated endpoints
+
+### 3.2 Data Storage Requirements
+
+#### User Profile Schema
+```json
+{
+  "userId": "uuid",
+  "email": "string",
+  "emailVerified": "boolean",
```

Comment:
Not needed. User accounts should not be created until a successful email auth happens, which means all user records should have verified emails because there's no other way to create an account.